internal/pkg/docker/dockerfile/dockerfile.go

```
@@ -0,0 +1,171 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+// Package dockerfile provides simple Dockerfile parsing functionality
+package dockerfile
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+var exposeRegexPattern = regexp.MustCompile(`(?P<port>\d+)(\/(?P<protocol>\w+))?`) // port and optional protocol, at least 1 time on a line
+
+const (
+	exposeRegexpWholeMatch = 0
+	exposeRegexpPort       = 1
+	exposeRegexpProtocol   = 3
+)
+const reFindAllMatches = -1 // regexp package uses this as shorthand for "find all matches in string"
+
+var (
+	errCouldntParseDockerfilePort = errors.New("parse port from EXPOSE")
+)
+
+type portConfig struct {
+	Port      uint16
+	Protocol  string
+	RawString string
+	err       error
+}
+
+// Dockerfile represents a parsed dockerfile
+type Dockerfile struct {
+	ExposedPorts []portConfig
+	parsed       bool
+	path         string
+
+	fs afero.Fs
+}
+
+// New returns an empty Dockerfile
+func New(fs afero.Fs, path string) *Dockerfile {
+	return &Dockerfile{
+		ExposedPorts: []portConfig{},
+		fs:           fs,
+		path:         path,
+		parsed:       false,
+	}
+}
+
+// GetExposedPorts returns a uint16 slice of exposed ports found in the Dockerfile
+func (df *Dockerfile) GetExposedPorts() ([]uint16, error) {
+	if !df.parsed {
+		err := df.parse()
+		if err != nil {
```

nit: Go has this neat syntax if you only care about errors:
```suggestion
		if err := df.parse(); err != nil {
```